entities/IoCContainer.java

```diff
@@ -0,0 +1,84 @@
+package entities;
+
+import java.nio.file.FileAlreadyExistsException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/*
+Lines for use the container
+1. IoCContainer ioc = IoCContainer.getInstance();
+2. ioc.Register(Entitie_name,new Entitie_name());
+3. Entitie_name any_name = (Entitie_name)ioc.Resolve("Entitie_name");
+4. Now you can use the name used in any_name for to the methodes of the entitie
+ */
+public class IoCContainer<T> {
+
+    //Singleton to instantiate the IoCContainer only once
+    private static IoCContainer ioc;
+    private IoCContainer(){}
+    public static IoCContainer getInstance(){
+        if(ioc == null){
+            ioc = new IoCContainer();
+        }return ioc;
+    }
+
+    //IoCContainer structure
+    private final Map<String, T> map = new HashMap<String,T>();
+
+    public void register(String s, T t) throws Exception {
+
+        Optional.ofNullable(s).orElseThrow(() -> new CannotBeNullException("Name cannot be null"));
+
+        Optional.ofNullable(map.get(s)).ifPresent(value -> {
+
+            try {
+                throw new AlreadyExistsException(s + "Already exists");
+            }catch (AlreadyExistsException e){
+                throw new RuntimeException(e);
+            }
+            });
+
+        map.put(s,t);
+    }
+
+    public T resolve(String s) throws Exception {
+        Optional<String> checkOpt = Optional.ofNullable(s);
+
+        Optional.ofNullable(s).orElseThrow(() -> new CannotBeNullException("Name cannot be null"));
+
+        Optional.ofNullable(map.get(s)).ifPresentOrElse(
+                value -> {
+
+                },
+                () -> {
+                    try {
+                        throw new DoesNotExists("Name cannot be null");
+                    }catch (DoesNotExists e){
+                        throw new RuntimeException(e);
+                    }
+        });
```

Comment:
Seem super complicated, try to handle each exception individually to improve code readability.

Also use java supliers whenever you can:
```
Optional.of(s).map(map::get).orElse(DoesNotExists::new);
```

Reply:
done